File: android-back-button-handling/MainActivity.java
// Java implementation
public class MainActivity extends AppCompatActivity {
    private WebView webView;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        webView = findViewById(R.id.webview);
        
        // Enable JavaScript and other WebView settings
        WebSettings webSettings = webView.getSettings();
        webSettings.setJavaScriptEnabled(true);
        webSettings.setDomStorageEnabled(true);
        
        // Load your web app
        webView.loadUrl("https://your-webapp-domain.com");
        
        // Set WebViewClient to handle navigation within WebView
        webView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                view.loadUrl(url);
                return true;
            }
        });
    }
    
    @Override
    public void onBackPressed() {
        // Check if WebView can go back
        if (webView.canGoBack()) {
            webView.goBack(); // Navigate back in web history
        } else {
            // Show confirmation dialog before closing app
            showExitConfirmationDialog();
        }
    }
    
    private void showExitConfirmationDialog() {
        new AlertDialog.Builder(this)
            .setTitle("Exit App")
            .setMessage("Are you sure you want to exit JSG Sparsh Portal?")
            .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    MainActivity.super.onBackPressed(); // Close the app
                }
            })
            .setNegativeButton("No", new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    dialog.dismiss(); // Stay in the app
                }
            })
            .show();
    }
}

File: android-back-button-handling/MainActivity.kt
// Kotlin implementation
class MainActivity : AppCompatActivity() {
    private lateinit var webView: WebView
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        
        webView = findViewById(R.id.webview)
        
        // Enable JavaScript and other WebView settings
        webView.settings.apply {
            javaScriptEnabled = true
            domStorageEnabled = true
            loadWithOverviewMode = true
            useWideViewPort = true
        }
        
        // Load your web app
        webView.loadUrl("https://your-webapp-domain.com")
        
        // Set WebViewClient to handle navigation within WebView
        webView.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
                url?.let { view?.loadUrl(it) }
                return true
            }
        }
    }
    
    override fun onBackPressed() {
        when {
            webView.canGoBack() -> {
                webView.goBack() // Navigate back in web history
            }
            else -> {
                showExitConfirmationDialog()
            }
        }
    }
    
    private fun showExitConfirmationDialog() {
        AlertDialog.Builder(this)
            .setTitle("Exit App")
            .setMessage("Are you sure you want to exit JSG Sparsh Portal?")
            .setPositiveButton("Yes") { _, _ ->
                super.onBackPressed() // Close the app
            }
            .setNegativeButton("No") { dialog, _ ->
                dialog.dismiss() // Stay in the app
            }
            .show()
    }
}